python/src/scipp/_variable.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @author Matthew Andrew
+from . import units, Unit, dtype, Variable
+from typing import Any, Sequence, Union
+import numpy
+
+
+def scalar(value: Any,
+           variance: Any = None,
+           unit: Unit = units.dimensionless,
+           dtype: type(dtype.float64) = None) -> Variable:
+    """Constructs a zero dimensional :class:`Variable` with a unit and optional
+    variance.
+
+    :param value: Initial value.
+    :param variance: Optional, initial variance, Default=None
+    :param unit: Optional, unit. Default=dimensionless
+    :param dtype: Optional, type of underlying data. Default=None,
+      in which case type is inferred from value input.
+      Cannot be specified for value types of
+      str, Dataset or DataArray.
+    :raises: Add this
+    """
+    if dtype is None:
+        return Variable(value=value, variance=variance, unit=unit)
+    else:
+        try:
+            return Variable(value=value,
+                            variance=variance,
+                            unit=unit,
+                            dtype=dtype)
+        except TypeError:
+            # Raise a more comprehensible error message in the case
+            # where a dtype cannot be specified.
+            raise TypeError(f"dtype cannot be specified for value"
+                            f" type of {type(value)}")
```

Comment:
How about `"Cannot convert {value.dtype} to {dtype}. Try omitting the 'dtype=' parameter."`?